src/main/java/com/fasterxml/jackson/databind/deser/BasicDeserializerFactory.java

```diff
@@ -508,12 +508,20 @@ private void _addSelectedPropertiesBasedCreator(DeserializationContext ctxt,
     {
         final int paramCount = candidate.paramCount();
         SettableBeanProperty[] properties = new SettableBeanProperty[paramCount];
+        boolean anySetterFound = false;
 
         for (int i = 0; i < paramCount; ++i) {
             JacksonInject.Value injectId = candidate.injection(i);
             AnnotatedParameter param = candidate.parameter(i);
             PropertyName name = candidate.paramName(i);
-            if (name == null) {
+            boolean isAnySetter = Boolean.TRUE.equals(ctxt.getAnnotationIntrospector().hasAnySetter(param));
+            if (isAnySetter) {
+                if (anySetterFound) {
```

Comment:
Looks like we lose information on any-setter, and only call introspector to find multiples here... but then need to re-introspect later on?

Reply:
Actually, the opposite of losing. At this point we are too early at introspection, where we have not constructed any getter yet.

I forgot to comment on this, but I could not find any appropriate way to contain the introspection result until the point where we actually try to construct any gutter

Reply:
I recollect this is somewhere in addFields() stage?

Reply:
I remembered now, this code serves two purposes.

- One is to detect and handle multiple any-setter (which, as you said, could be moved to later one below)
- The other one, which I do need help with is... not any-setter not having any property name here. So any-setter creator parameter without any name specified (via `@JsonProperty`) throws error.

Reply:
LMK if you have better idea with the second problem I am facing, @cowtowncoder

Reply:
Ah. Yes. For that purpose, we could only do a lookup if `name` not found. Let me think about this a bit, good point, something is needed here.